src/transform/schema.ts

```diff
@@ -28,12 +31,22 @@ export function transformSchemaObjMap(obj: Record<string, any>, options: Transfo
     if (value.description) output += comment(value.description);
 
     // 2. name (with “?” if optional property)
-    output += `${readonly}"${key}"${required.includes(key) ? "" : "?"}: `;
+
+    let hasDefault = false;
+    if (value.schema != null) {
+      const schema = resolveRefIfNeeded<SchemaObject>(options.document, value.schema);
```

Comment:
Not a request for change, but is passing the entire document necessary here? Could we simply add refs that have default values to that `required[]` array? That way we’re figuring this stuff out in the context that has that information, and not at the schema object level. But happy for pushback if that‘s simply not possible.

Reply:
The default value is set on the actual values themselves, as such it seems that the PR follows most closely to your goal.  I think manipulating the required array would certainly be not as clear.

Additionally, if the required array isn't specified there would be special code to handle that.

Reply:
If it’s a problem of the `required` array being nullable, we could fix that. From what I hear you saying, it would be possible to modify the `required` array. If that’s the case, I vote for doing that instead. I disagree that this is “clearer;“ this PR does take a module which previously was only worried about a single schema object at a time, and now this module has complete document context and so now its responsibilities are less clear.

Also, as an aside, this library does support [partial schemas](https://github.com/drwpow/openapi-typescript/#cli-options), so I’d like to avoid handling that logic in two places (here and at the top level) if possible.

If it’s simply an issue of `required` being too unclear a name, I vote for renaming that array. Or have it be an array with more data in there about default properties so that we don’t need to tunnel the document through multiple levels of modules.